mllib/src/test/scala/org/apache/spark/ml/util/StopwatchSuite.scala

```diff
@@ -29,18 +33,27 @@ class StopwatchSuite extends SparkFunSuite with MLlibTestSparkContext {
     intercept[AssertionError] {
       sw.stop()
     }
+    val ubStart = now
     sw.start()
-    Thread.sleep(50)
+    val lbStart = now
+    runTask()
+    val lb = now - lbStart
     val duration = sw.stop()
-    assert(duration >= 50 && duration < 100) // using a loose upper bound
+    val ub = now - ubStart
+    assert(duration >= lb && duration <= ub)
     val elapsed = sw.elapsed()
     assert(elapsed === duration)
+    val ubStart2 = now
```

Comment:
It might be nice to create a helper method, rather than the now-complex timing code which gets repeated for each start/stop section in these tests.